CMakeLists.txt.in

```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 2.8.2)
+
+project(googletest-download NONE)
+
+include(ExternalProject)
+ExternalProject_Add(googletest
+  GIT_REPOSITORY    https://github.com/google/googletest.git
+  GIT_TAG           master
+  SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-src"
```

Comment:
>googletest [](start = 49, length = 10)

I don't have huge concern over using gtest. But please confirm if all the C++ projects are also forced to use gtest :).

Reply:
We probably want to use a single test framework for ease of building, documentation, and CI setup across the whole repo. If you have reasons to pick another, we should discuss the trade offs and let's pick one that meets the necessary requirements.

Reply:
There is not a final decision about that test framework we have to use.
I added this instead of expending time to manually create assert macros that eventually might be replaced.
I think we would talk about it and decide if we keep gtest or any other

Reply:
So far, I like that we can integrate gtest directly with cmake. This makes a seamless CI integration and easy-to-get dependency.
And it's not too bad to use as well.